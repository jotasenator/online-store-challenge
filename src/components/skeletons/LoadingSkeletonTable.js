import React from 'react'
import Skeleton from 'react-loading-skeleton';



export default function LoadingSkeletonTable() {
    return (
        <div className='container mt-5 mb-5 pt-5 pb-5 '>

            <div className='head-table-skeleton'>
                <div className='test-skeleton'><Skeleton width={80} height={40} /></div>
                <div><Skeleton width={80} height={40} /></div>
                <div><Skeleton width={80} height={40} /></div>
                <div><Skeleton width={80} height={40} /></div>
                <div><Skeleton width={80} height={40} /></div>
                <div><Skeleton width={80} height={40} /></div>
                <div><Skeleton width={80} height={40} /></div>
            </div>

            <Skeleton count={10} height={70} />
        </div>
    )
}
