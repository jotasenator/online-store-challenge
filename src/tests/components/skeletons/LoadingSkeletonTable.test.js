import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'


import LoadingSkeletonTable from '../../../components/skeletons/LoadingSkeletonTable'

describe('LoadingSkeletonTable test', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<LoadingSkeletonTable />);
    });

    afterEach(() => {
        wrapper.unmount();
    });


    test('should render <LoadingSkeletonTable/> correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
    test('should show me what is inside className=container', () => {
        expect(wrapper.find('.container').text()).toBe('<Skeleton /><Skeleton /><Skeleton /><Skeleton /><Skeleton /><Skeleton /><Skeleton /><Skeleton />')
    })
    test('should show me what is inside className=head-table-skeleton', () => {
        expect(wrapper.find('.head-table-skeleton').text()).toBe('<Skeleton /><Skeleton /><Skeleton /><Skeleton /><Skeleton /><Skeleton /><Skeleton />')
    })

})
