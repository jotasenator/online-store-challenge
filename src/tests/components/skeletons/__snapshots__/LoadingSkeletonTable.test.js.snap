// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSkeletonTable test should render <LoadingSkeletonTable/> correctly 1`] = `
<div
  className="container mt-5 mb-5 pt-5 pb-5 "
>
  <div
    className="head-table-skeleton"
  >
    <div
      className="test-skeleton"
    >
      <Skeleton
        circle={false}
        count={1}
        delay={0}
        duration={1.2}
        height={40}
        width={80}
        wrapper={null}
      />
    </div>
    <div>
      <Skeleton
        circle={false}
        count={1}
        delay={0}
        duration={1.2}
        height={40}
        width={80}
        wrapper={null}
      />
    </div>
    <div>
      <Skeleton
        circle={false}
        count={1}
        delay={0}
        duration={1.2}
        height={40}
        width={80}
        wrapper={null}
      />
    </div>
    <div>
      <Skeleton
        circle={false}
        count={1}
        delay={0}
        duration={1.2}
        height={40}
        width={80}
        wrapper={null}
      />
    </div>
    <div>
      <Skeleton
        circle={false}
        count={1}
        delay={0}
        duration={1.2}
        height={40}
        width={80}
        wrapper={null}
      />
    </div>
    <div>
      <Skeleton
        circle={false}
        count={1}
        delay={0}
        duration={1.2}
        height={40}
        width={80}
        wrapper={null}
      />
    </div>
    <div>
      <Skeleton
        circle={false}
        count={1}
        delay={0}
        duration={1.2}
        height={40}
        width={80}
        wrapper={null}
      />
    </div>
  </div>
  <Skeleton
    circle={false}
    count={10}
    delay={0}
    duration={1.2}
    height={70}
    width={null}
    wrapper={null}
  />
</div>
`;
